addons/addon-base-raas/packages/base-raas-workflow-steps/lib/steps/provision-environment/provision-environment.js

```diff
@@ -141,6 +141,15 @@ class ProvisionEnvironment extends StepBase {
       addParam('AmiId', amiImage);
       addParam('KeyName', keyName);
     }
+
+    if (type === 'ec2-windows' || type === 'ec2-linux') {
+      addParam('SolutionNamespace', this.settings.get('solutionNamespace'));
+    }
+
+    if (type === 'sagemaker') {
+      addParam('IsAppStreamEnabled', this.settings.get('isAppStreamEnabled').toString());
+    }
```

Comment:
is there a possibility that these parameters will be added to an old template and not to the new template that customers haven't imported yet. Can we fail environment provisioning in case the template is old and not suited for AppStream but the solution has it enabled?

Reply:
Turns out this `provision-environment` file was used for built in workspaces and not SC workspaces. I made the updates in the correct file, which is `env-config-var-service`. I tested this, and even if we pass these parameters to an old template, the workspace still gets started correctly. 

In terms of checking whether an environment is launching the right type of workspace, I think that check should be done earlier in the process than when the workflowLoopRunner is started. When a user switch from normal workspaces to AppStream enabled workspace, we should have a post deployment script that disables all of the "normal" workspace types. And then from then on, we should check that a workspace type is AppStream compatible before letting the user approves the workspace for launch.